client.go

```diff
@@ -249,9 +249,13 @@ func (c *Client) String() string {
 func (c *Client) Clone(opts ...ClientOption) *Client {
 	options := c.options.Clone()
 	options.Apply(opts...)
-	return &Client{
+	clone := &Client{
 		options: options,
 	}
+	if err := clone.setup(); err != nil {
+		return nil
+	}
```

Comment:
Maybe it's better to just ignore the error here. If you Clone() a broken connection, you get a broken connection from Clone() too.